tests/integration_tests/test_pytorch_distributed.py

```diff
@@ -351,3 +351,20 @@ def test_updates_properties(storage_mode: str) -> None:
             [getattr(trial, p) for p in property_names]
 
         dist.barrier()  # type: ignore[no-untyped-call]
+
+
+@pytest.mark.filterwarnings("ignore::optuna.exceptions.ExperimentalWarning")
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_pass_frozen_trial_to_torch_distributed(storage_mode: str) -> None:
+    # Regression test of #4697
+    def objective(trial: optuna.trial.BaseTrial) -> float:
+        trial = optuna.integration.TorchDistributedTrial(trial if dist.get_rank() == 0 else None)
+        x = trial.suggest_float("x", low=-100, high=100)
+        return x * x
+
+    with StorageSupplier(storage_mode) as storage:
+        study = optuna.create_study(direction="minimize", storage=storage)
+        study.optimize(objective, n_trials=1)
+        best_trial = study.best_trial
+
+        objective(best_trial)
```

Comment:
We may want to check the re-evaluation result as follows, but it is optional. Let me work on it in a follow-up PR.
```suggestion
        actual = objective(best_trial)
        assert best_trial.value == actual
```

Reply:
Thanks!